Add initial project file to build with homebrew environment

utzig opened the pull request:
For Mac users this enables building against homebrew installed versions of ffmpeg, freetype2, qt5 and qwt. The generated .app is not yet working due to the standard homebrew installation of qwt linking it to qt4, but I'll send a patch to them. Also I did some local modifications for not requiring blackmagic sdk and mocked some ffmpeg_glue calls due to not being completely compatible with homebrew version of ffmpeg. This should be easy to fix and I'll send a patch later.

utzig commented:
With the updates it should now be able to build with Homebrew. Follow the README.

utzig commented on Project/Homebrew/qctools.pro:
```diff
@@ -0,0 +1,95 @@
+
+QT_CONFIG -= no-pkg-config
+
+include ( $$system(brew --prefix qwt)/features/qwt.prf )
```
I just used this .project to build for Linux. The only line that gives any problem is this one which is dependent on homebrew. Should probably add some heuristics here checking for homebrew if on mac, and common paths for qwt prf's files and if not found quit with error message.

JeromeMartinez commented:
Homebrew/qctools.pro: I am reluctant to have 2 different .pro files. Isn't the a solution for having an option in the original .pro file for adding what you need for brew?

FFmpeg_Glue.cpp: we need such info for logging which FFmeg version was in use. OK the way it is currently implemented is wrong (not in the system package), but I guess there is another method for getting such info from FFmpeg API. I'll check.

dericed commented on Project/Homebrew/README.md:
```diff
@@ -0,0 +1,22 @@
+require not having a homebrew installation of qt (aka qt4):
+
+    $ brew uninstall qwt
+    $ brew install ./qwt
```
`brew install ./qwt` doesn't work. But `brew install qwt.rb` does.

dericed commented on Project/Homebrew/README.md:
```diff
@@ -0,0 +1,22 @@
+Install ffmpeg, freetype2, qt5
+
+    $ brew install ffmpeg freetype2 qt5
+
```
should mention that sometimes `brew link --force qt5` is needed, since homebrew normally doesn't link qt5

utzig replied:
Oh, I don't remember having to do that but I only have Qt5 on my homebrew. Btw, I will suggest adding a qwt-qt5 to homebrew like Linux distros do because I don't think it's possible to build only a single library that links against two different Qt frameworks. I'll open a pull request there in the following days and see what the maintainers think about it.

dericed replied:
[qt5 is keg_only](https://github.com/Homebrew/homebrew/blob/master/Library/Formula/qt5.rb#L64) so is not linked by default. Some doc on this is here; https://github.com/Homebrew/homebrew/blob/15f291ddaa05cc010456a733b799382175c29dd4/share/doc/homebrew/How-to-build-software-outside-Homebrew-with-Homebrew-keg-only-dependencies.md#how-do-i-get-non-homebrew-builds-to-find-those-tools.

dericed commented:
`qmake && make` built the app properly in my mac/brew environment. However when I open a file in qctools, and then select Export>"Sidecar xml.gz" the app crashes. As Jerome notes the edits to FFmpeg_glue may break the export.

utzig commented:
Jerome,

I think it's reasonable to have a single build system but I don't think it is easy to do with the qt build system. I will work on a CMake based build which can cover all different environment options but I don't think that it should be a priority right now. Maybe I'll open an issue just to remember about it.

Regarding ffmpeg glue, I don't think the information about compiler/year/etc are really available with the standard API but I'll check. Also the issue with exporting the XML I should be able to fix today or in the following days.

utzig commented:
Ok, fixed the glue problems which were related to wrong type in returns. I was reading the ffprobe code and it does pretty much the same that is done by QCTools which is reading from config.h. I guess the only way to get the data would be to run ffprobe --version and capture the output. For the sake of having something to add to the XML fields, I'm returning the current year (which of course might not be the one when ffprobe was built!) and "not available" for the compiler, etc information.

I changed the qwt formula which is now called qwt-qt5. This enables having both qwt (linked for Qt 4) and qwt-qt5 which is linked against Qt 5. This will require reinstallation:

    $ brew uninstall qwt
    $ brew install qwt (*optional*)
    $ brew install qwt-qt5.rb

And regenerating the Makefile with:

    $ $(brew --prefix qt5)/bin/qmake qctools.pro

I submitted the Formula to the Homebrew project. Let's see how it goes!

dericed commented:
merging, I'll add an issue based on Jerome's comment for later.
